format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatDataFormatter.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.xls.input;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.DateUtil;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+
+/**
+ * Formats the cell value of an Excel file.
+ */
+public class XlsInputFormatDataFormatter {
+  private static final DataFormatter dataFormatter = new DataFormatter();
+  private final FormulaEvaluator evaluator;
+
+  /**
+   * Constructor for XlsInputFormatDataFormatter.
+   *
+   * @param evaluator the formula evaluator
+   */
+  public XlsInputFormatDataFormatter(FormulaEvaluator evaluator) {
+    this.evaluator = evaluator;
+  }
+
+  /**
+   * Formats the cell value of an Excel file.
+   *
+   * @param cell the cell to format
+   * @return the formatted cell value
+   */
+  public String formatCellValue(Cell cell, Schema.Type type) {
+    if (cell == null) {
+      return null;
+    }
+
+    CellType cellType = cell.getCellType();
+    if (cellType == CellType.FORMULA) {
+      try {
+        cellType = cell.getCachedFormulaResultType();
+      } catch (Exception e) {
+        cellType = evaluator.evaluateFormulaCell(cell);
+      }
+    }
+
+    switch (cellType) {
+      case NUMERIC:
+        if (DateUtil.isCellDateFormatted(cell)) {
+          return dataFormatter.formatCellValue(cell);
+        }
+        return Double.toString(cell.getNumericCellValue());
+      case STRING:
+        if (type == Schema.Type.DOUBLE) {
+          // Edge case when schema was inferred as double but the cell is actually a string
+          // this can be caused by an error formula cell, as the error value is stored as a string
+          return null;
+        }
+        return cell.getRichStringCellValue().getString();
+      case BOOLEAN:
+        return cell.getBooleanCellValue() ? "TRUE" : "FALSE";
+      case BLANK:
+      case ERROR:
+        return null;
+      default:
+        throw new IllegalStateException(String.format("Unexpected celltype (%s)", cellType));
```

Comment:
Elaborate the message by specifying the list of Supported types.

Reply:
This is an impossible state , only possible if `xls` has a new cell type added in future, this is internal representation of cell by library we are using. This is an implementation details not relevant for end user.

Reply:
Even in impossible state we should be informative about the error messages. So, lets rewrite the error messages something as:
`Failed to format (%s) due to unsupported cell type (%s)`.